lib/coinbase/transfer.rb

```diff
@@ -134,26 +134,14 @@ def transaction
     # Returns the status of the Transfer.
     # @return [Symbol] The status
     def status
-      # Check if the transfer has been signed yet.
-      return Status::PENDING if transaction_hash.nil?
-
-      onchain_transaction = Coinbase.configuration.base_sepolia_client.eth_getTransactionByHash(transaction_hash)
-
-      if onchain_transaction.nil?
-        # If the transaction has not been broadcast, it is still pending.
-        Status::PENDING
-      elsif onchain_transaction['blockHash'].nil?
-        # If the transaction has been broadcast but hasn't been included in a block, it is
-        # broadcast.
-        Status::BROADCAST
-      else
-        transaction_receipt = Coinbase.configuration.base_sepolia_client.eth_getTransactionReceipt(transaction_hash)
+      @model.status
+    end
 
-        if transaction_receipt['status'].to_i(16) == 1
-          Status::COMPLETE
-        else
-          Status::FAILED
-        end
+    # Reload reloads the Transfer model with the latest version from the server side.
+    # @return [Transfer] The most recent version of Transfer from the server.
+    def reload
+      @model = Coinbase.call_api do
+        transfers_api.get_transfer(wallet_id, from_address_id, id)
       end
```

Comment:
We should make sure that we return self at the end of `reload`, otherwise we will return `@model`. We should probably add a test that we're returning a `Coinbase::Transfer`.

I believe I added this incorrectly in the wallet / address file

Reply:
added `expect(transfer).to be_a(Coinbase::Transfer)` to reload tests